contributors/design-proposals/auth/validated-pod-annotations.md

```diff
@@ -0,0 +1,116 @@
+# Validated Pod Annotations
+
+Author: @mikedanese
+
+# Objective
+
+Propose a mechanism to enable adminstrators to apply extra authorization checks
+on operations that grant direct or indirect access to a pods runtime
+environment.
+
+# Background
+
+A number of pod identity integrations need to inject credentials from external
+identity providers into a pod runtime environment. A common and desirable
+property of an identity solutions is:
+
+> An actor should be able to access credentials of an identity only if the actor
+> has explicit authority to act as that identity.
+
+Today, naive pod identity integrations have weaker guarantees. To illustrate the
+issue, suppose a pod is running with access to a kerberos credential injected
+with a flex volume. A user can access the kerberos credential if the user can do
+any of the following:
+
+1. Exec into the pod.
+1. Update the image of the pod to run custom code to export the credential.
+1. Update the image in the PodSpec of any controller that is in the owner
+   hierarchy of the pod.
+1. Create another pod that uses the same credential and runs custom code to
+   export the credential.
+1. Create a controller that then creates a pod that uses the same credential and
+   runs custom code to export the credential.
+
+Ideally, a user can only access the kerberos credential if the user can:
+
+1. Act as the identity.
+
+This constraint allows a security auditor to audit identity access with only the
+delegation policy. Identity integrations can work around this by implementing
+custom admission controllers but the issue is common and general enough to
+warrent a consisntent API (and thus a requires a solution in core).
+
+# Proposal
+
+We can add a "validatedAnnotations" field on PodSpec and validate these
```

Comment:
I agree the most compelling use case is on the `PodSpec`. Are there any use cases for this beyond things that end up creating Pods? Could this be useful alongside the standard annotations in `ObjectMeta`?

Reply:
I'm not sure that would solve this use case. Annotations aren't passed to sub-objects.

Reply:
I agree. There may be some CRDs that would benefit from this as well (e.g. Istio service authn/authz policies that need to be bound to the same service identity).